service-registry/registry-consul/src/main/java/org/apache/servicecomb/registry/consul/ConsulRegistration.java

```diff
@@ -0,0 +1,176 @@
+
+  @Override
+  public String name() {
+    return ConsulConst.CONSUL_REGISTRY_NAME;
+  }
+
+  @Override
+  public ConsulRegistrationInstance getMicroserviceInstance() {
+    return new ConsulRegistrationInstance(consulInstance);
+  }
+
+  @Override
+  public boolean updateMicroserviceInstanceStatus(MicroserviceInstanceStatus status) {
+    return true;
+  }
+
+  @Override
+  public void addSchema(String schemaId, String content) {
+    if (consulDiscoveryProperties.isEnableSwaggerRegistration()) {
+      consulInstance.addSchema(schemaId, content);
+    }
+  }
+
+  @Override
+  public void addEndpoint(String endpoint) {
+    consulInstance.addEndpoint(endpoint);
+  }
+
+  @Override
+  public void addProperty(String key, String value) {
+    consulInstance.addProperty(key, value);
+  }
+
+  @Override
+  public boolean enabled() {
+    return consulDiscoveryProperties.isEnabled();
+  }
+
+  @Override
+  public void init() {
+    logger.info("ConsulRegistration init");
+    String serverPort;
+    if (restAddress.contains("?")) {
+      serverPort = restAddress.substring(restAddress.indexOf(":") + 1, restAddress.indexOf("?"));
+    } else {
+      serverPort = restAddress.substring(restAddress.indexOf(":") + 1);
+    }
+    String env = BootStrapProperties.readServiceEnvironment(environment);
+    if (StringUtils.isEmpty(env)) {
+      env = ConsulConst.CONSUL_DEFAULT_ENVIRONMENT;
+    }
+    String instanceId = registrationId.getInstanceId();
+    String serviceName = BootStrapProperties.readServiceName(environment);
+    consulInstance = new ConsulInstance();
+    consulInstance.setServiceId(serviceName + "-" + serverPort);
+    consulInstance.setInstanceId(instanceId);
+    consulInstance.setEnvironment(env);
+    consulInstance.setApplication(BootStrapProperties.readApplication(environment));
+    consulInstance.setServiceName(BootStrapProperties.readServiceName(environment));
+    consulInstance.setAlias(BootStrapProperties.readServiceAlias(environment));
+    consulInstance.setDescription(BootStrapProperties.readServiceDescription(environment));
+    if (StringUtils.isNotEmpty(dataCenterProperties.getName())) {
+      DataCenterInfo dataCenterInfo = new DataCenterInfo();
+      dataCenterInfo.setName(dataCenterProperties.getName());
+      dataCenterInfo.setRegion(dataCenterProperties.getRegion());
+      dataCenterInfo.setAvailableZone(dataCenterProperties.getAvailableZone());
+      consulInstance.setDataCenterInfo(dataCenterInfo);
+    }
+    consulInstance.setProperties(BootStrapProperties.readServiceProperties(environment));
+    consulInstance.setVersion(BootStrapProperties.readServiceVersion(environment));
+
+    List<String> tags = new ArrayList<>();
+    tags.add(consulInstance.getServiceName());
+    registrationBuilder = ImmutableRegistration.builder()
+        .id(consulInstance.getServiceId())
+        .name(consulInstance.getServiceName())
+        .address(consulDiscoveryProperties.getIpAddress())
```

Comment:
See zookeeper/nacos implementation how to get ip address and port. Do not use the new InetUtils added. InetUtils implementation is not correct.